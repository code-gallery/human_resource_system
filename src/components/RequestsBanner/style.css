.RequestsBanner {
  box-sizing: border-box;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 100%;
  height: 45px;
  overflow: hidden;
  padding: 0 15px;
  background-color: #72d371;
}

.RequestsBannerMobile {
  box-sizing: border-box;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 100;
  display: flex;
  justify-content: center;
  align-items: center;
  min-width: 100%;
  height: 50px;
  overflow: hidden;
  padding: 4px 15px;
  background-color: #f54f4f;

}
.RequestsBanner__msg {
  margin: 0;
  font-family: 'Graphik-Regular', sans-serif;
  font-size: 13px;
  text-align: center;
  color: #fff;
}

@media (min-width: 768px) {
  .RequestsBanner__msg {
    font-size: 16px;
  }
}
