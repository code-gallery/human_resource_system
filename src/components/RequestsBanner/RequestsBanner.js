import React from 'react'
import PropTypes from 'prop-types'
import './style.css'

const RequestsBanner = ({ requests, className }) => {
  if (!Array.isArray(requests)) {
    return null
  }
const mobileLink={
  color:'#fff',
  fontWeight:'bold'
}


  let incompleteRequests = requests
    .filter(({ status }) => status == 'pending')

    let filteredReq = []
    for (let i = 0; i < incompleteRequests.length; i++) {
      let req = incompleteRequests[i]
      let checks = req.checks
      let filteredChecks = checks.filter((item) => item.options && item.options.checkForMobile === undefined)
      if (filteredChecks.length > 0) {
        req.checks = filteredChecks
        filteredReq.push(req)
      }
    }
  
    incompleteRequests = [...filteredReq]

  if (!incompleteRequests.length) {
    return null
  }

  if (incompleteRequests.length === 1) {
    const name = incompleteRequests[0].candidate.organisation.name
    
    return (
      <React.Fragment>
      { window.innerWidth <= 768 && <aside className={`RequestsBannerMobile ${className}`}>
       <p className="RequestsBanner__msg">
        CLICK HERE TO <a href="https://appappii.page.link/mVFa" style={mobileLink}>DOWNLOAD APPII APP</a> to complete checks.
       </p>
      </aside>
     }
     { window.innerWidth > 768 && <aside className={`RequestsBanner ${className}`}>
       <p className="RequestsBanner__msg">
        You have an incomplete Work Pass Request from {name}.
        Please use the APPII mobile app to complete them.
       </p>
      </aside>
    }
    </React.Fragment>
    )
  }

  return (
    <React.Fragment>
      { window.innerWidth <= 768 &&
      <aside className={`RequestsBannerMobile ${className}`}>
      <p className="RequestsBanner__msg">
        CLICK HERE TO <a href="https://appappii.page.link/mVFa" style={mobileLink}>DOWNLOAD APPII APP</a> to complete checks.
      </p>
    </aside>
    }
    {window.innerWidth > 768 &&
    <aside className={`RequestsBanner ${className}`}>
      <p className="RequestsBanner__msg">
        You have <b>{incompleteRequests.length} incomplete Work Pass Requests</b>.
        Please use the APPII mobile app to complete them.
      </p>
    </aside>
    }
   </React.Fragment>
  )
}

RequestsBanner.propTypes = {
  requests: PropTypes.array.isRequired,
  className: PropTypes.string
}

RequestsBanner.defaultProps = {
  className: ''
}

export default RequestsBanner
