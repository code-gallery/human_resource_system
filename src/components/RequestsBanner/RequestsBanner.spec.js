import React from 'react'
import renderer from 'react-test-renderer'
import RequestsBanner from './RequestsBanner'

describe('<RequestsBanner />', () => {
  const commonProps = {
    requests: [
      {
        id: 1,
        candidate: {
          organisation: { name: 'Applied Blockchain' }
        }
      }
    ]
  }

  describe('Rendering', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <RequestsBanner {...commonProps} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    it('renders correctly for 2 requests', () => {
      const props = {
        requests: [
          ...commonProps.requests,
          {
            id: 2,
            candidate: {
              organisation: { name: 'Shell' }
            }
          }
        ]
      }
      const tree = renderer.create(
        <RequestsBanner {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
