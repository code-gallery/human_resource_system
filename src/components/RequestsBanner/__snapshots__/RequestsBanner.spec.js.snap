// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestsBanner /> Rendering renders correctly 1`] = `
<aside
  className="RequestsBanner "
>
  <p
    className="RequestsBanner__msg"
  >
    You have an incomplete Work Pass Request from 
    Applied Blockchain
    . Please use the APPII mobile app to complete it.
  </p>
</aside>
`;

exports[`<RequestsBanner /> Rendering renders correctly for 2 requests 1`] = `
<aside
  className="RequestsBanner "
>
  <p
    className="RequestsBanner__msg"
  >
    You have 
    <b>
      2
       incomplete Work Pass Requests
    </b>
    . Please use the APPII mobile app to complete them.
  </p>
</aside>
`;
